syft/pkg/cataloger/java/archive_parser.go

```diff
@@ -433,14 +434,53 @@ func artifactIDMatchesFilename(artifactID, fileName string, artifactsMap map[str
 	if artifactID == "" || fileName == "" {
 		return false
 	}
-	// Ensure true is returned when filename matches the artifact ID, prevent random retrieval by checking prefix and suffix
+
+	// First, try exact match
+	if artifactID == fileName {
+		return true
+	}
+
+	// If there's an exact match in the artifacts map, use that
 	if _, exists := artifactsMap[fileName]; exists {
 		return artifactID == fileName
 	}
-	// Use fallback check with suffix and prefix if no POM properties file matches the exact artifact name
+
+	// Enhanced matching logic: try to match artifactID with filename more intelligently
+	// This handles cases where the filename might have version suffixes or other variations
+
+	// Remove common version patterns from both for comparison
+	cleanArtifactID := removeVersionSuffix(artifactID)
+	cleanFileName := removeVersionSuffix(fileName)
+
+	// Try exact match after cleaning
+	if cleanArtifactID == cleanFileName {
+		return true
+	}
+
+	// Try matching with the original logic as fallback
 	return strings.HasPrefix(artifactID, fileName) || strings.HasSuffix(fileName, artifactID)
 }
 
+// removeVersionSuffix removes common version patterns from artifact names to improve matching
+func removeVersionSuffix(name string) string {
+	// Remove patterns like: -1.2.3, _1.2.3, .1.2.3
+	versionPatterns := []string{
+		`-\d+(\.\d+)*.*$`,
```

Comment:
This could be consolidated to a single pattern using the character set: `[-_.]`, e.g. `[-_.]\d+(\.\d+)*$` and I've removed the `.*` at the end will result in it matching things like `spring-2something`, which doesn't seem intended. I think we're going to want to be very careful making changes here, while this Java JAR determination logic may seem confusing, it's been refined a lot over the years for many specific cases.

I think we should add a test for something like the Kafka jars: https://repo1.maven.org/maven2/org/apache/kafka/kafka_2.10/0.10.2.0/ -- note these have `_2.10`, which is part of the package name, followed by a version of `-0.10.2.0`. Using the regexes defined in this PR, `_2.10-0.10.2.0` would match the version, whereas adjusting it as I've noted in this comment would not.